src/nvim/eval.c

```diff
@@ -11841,6 +11931,830 @@ static void f_mode(typval_T *argvars, typval_T *rettv)
   rettv->v_type = VAR_STRING;
 }
 
+/// Msgpack callback for writing to readfile()-style list
+static int msgpack_list_write(void *data, const char *buf, size_t len)
```

Comment:
Perhaps we should create a `src/nvim/eval/` subdirectory that contains eval sub-modules like this, and this could live in `eval/msgpack.c`. Or `eval_msgpack.c` at the top level is fine too.